server/src/main/java/org/opensearch/search/profile/ProfileMetric.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.profile;
+
+import org.opensearch.common.annotation.PublicApi;
+
+import java.util.Map;
+
+/**
+ * A metric for profiling.
+ */
+@PublicApi(since = "1.0.0")
+public abstract class ProfileMetric {
```

Comment:
It looks like `ProfileMetric` has to have a type class (Timer, ...), so we won't be typecasting more than necessary